### PR TITLE
Import Durable Object classes in type generation

### PR DESCRIPTION
**Pull request by ottomated**
## What this PR solves / how to test

Adds type safety to Durable Objects in type generation.

Old:
```ts
interface Env {
  DO: DurableObjectNamespace;
}
```

New:
```ts
interface Env {
  DO: DurableObjectNamespace<import("./src/index").MyDurableObject>;
}
```

Import specifier is generated relative from the output file to the worker entrypoint.

## Author has addressed the following

- Tests
  - [x] Included
- E2E Tests CI Job required? (Use "e2e" label or ask maintainer to run separately)
  - [x] I don't know
- Changeset ([Changeset guidelines](https://github.com/cloudflare/workers-sdk/blob/main/CONTRIBUTING.md#changesets))
  - [x] Included
- Public documentation
  - [x] Not necessary because: doesn't require any user action to benefit

### CODE REVIEW

File: packages/wrangler/src/__tests__/type-generation.test.ts [hunk 1]
```diff
@@ -112,6 +131,7 @@ const bindingsConfigMock: Omit<
 		bindings: [
 			{ name: "DURABLE_TEST1", class_name: "Durability1" },
 			{ name: "DURABLE_TEST2", class_name: "Durability2" },
+			{ name: "DURABLE_TEST3", class_name: "Durability3" },
```
Comment (petebacondarwin) [inline packages/wrangler/src/__tests__/type-generation.test.ts:134]: What about a Durable Object defined in another Worker?
Can you add a test to ensure that this works correctly?
Particularly in the case where there is a matching file in the local Worker which may be picked up incorrectly.

**Review by petebacondarwin — CHANGES_REQUESTED** [review]
Thanks for this. Please can you check and test the external DO use case.

File: packages/wrangler/src/__tests__/type-generation.test.ts (hunk 1 above)
Comment (ottomated) [reply packages/wrangler/src/__tests__/type-generation.test.ts]: Test added!

File: packages/wrangler/src/__tests__/type-generation.test.ts [hunk 2]
```diff
@@ -286,8 +319,10 @@ describe("generateTypes()", () => {
 			ANOTHER: \\"thing\\";
 			\\"some-other-var\\": \\"some-other-value\\";
 			OBJECT_VAR: {\\"enterprise\\":\\"1701-D\\",\\"activeDuty\\":true,\\"captian\\":\\"Picard\\"};
-			DURABLE_TEST1: DurableObjectNamespace;
-			DURABLE_TEST2: DurableObjectNamespace;
+			DURABLE_DIRECT_EXPORT: DurableObjectNamespace<import(\\"./index\\").DurableDirect>;
+			DURABLE_RE_EXPORT: DurableObjectNamespace<import(\\"./index\\").DurableReexport>;
+			DURABLE_NO_EXPORT: DurableObjectNamespace /* DurableNoexport */;
+			DURABLE_EXTERNAL: DurableObjectNamespace /* DurableExternal from external-worker */;
```
Comment (petebacondarwin) [inline packages/wrangler/src/__tests__/type-generation.test.ts:325]: 🎉

**Review by petebacondarwin — APPROVED** [review]

**Pull request merged**